pkg/docker/config/config.go

```diff
@@ -0,0 +1,271 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/types"
+	helperclient "github.com/docker/docker-credential-helpers/client"
+	"github.com/docker/docker/pkg/homedir"
+	"github.com/pkg/errors"
+)
+
+type dockerAuthConfig struct {
+	Auth string `json:"auth,omitempty"`
+}
+
+type dockerConfigFile struct {
+	AuthConfigs map[string]dockerAuthConfig `json:"auths"`
+	CredHelpers map[string]string           `json:"credHelpers,omitempty"`
+}
+
+const (
+	authCfg           = ".config/containers"
+	authCfgFileName   = "auth.json"
+	dockerCfg         = ".docker"
+	dockerCfgFileName = "config.json"
+)
+
+var (
+	// ErrNotLoggedIn is returned for users not logged into a registry
+	// that they are trying to logout of
+	ErrNotLoggedIn = errors.Errorf("not logged in")
+)
+
+// SetAuthentication stores the base64 encoding of username and password
+// in the auth file if the credentials are authenticated
+func SetAuthentication(username, password, registry string) error {
+	path, err := getPathToAuth()
+	if err != nil {
+		return errors.Wrapf(err, "error getting path to auth file")
+	}
+	encodedCreds := encodeUserAndPass(username, password)
+	return addAuthToJSON(registry, encodedCreds, path)
```

Comment:
Perhaps `addAuthToJSON` could be inlined here.  Not sure, depends on how the integration of credential helpers will impact the code.